### ISSUE TITLE
main_exploratory_analysis

### ISSUE THREAD

**Issue by angelinetsui** [manybabies/mb1-cdi-followup#24]
Hi Luis, I will start reviewing your codes in 05_main_exploratory, but I think before I do a review, I will need more information about why you complete certain steps and etc. A few suggestions:

1) Please explain why we need to check the distributions of IV (one to two sentences will be fine), is this related to the assumptions in the beta-regression models? 

2) Do we have assumptions for the beta-regression models (for example, there are codes about stability of models, may I ask why we need to check it)? If you have documentation about beta-regression, can you include links on the 05_main_exploratory?

3) I cannot remember our previous discussion about which model that we want to report in the manuscript. Right now, we have two models in the 05 Rmd: Full and pruned. Which one we should report? Please explain in the 05 Rmd that why we need to run two models, which one is used in the final report and why? 

I may have more questions, but for now, these are the questions that I have. Can you please add some more explanations to the 05 Rmd, and I will take a look after you have done this? Thanks @munozalvalu

**Comment by munozalvalu** [comment]
Hi Angeline,

Thank you for contacting me! I've updated the 05_main_exploratory file with responses to your questions. Please let me know if there's anything else to be clarified.

**Comment by angelinetsui** [comment]
Hi Luis, great, thanks for the reply. Do you have any reference (e.g., links) that talks about beta regression? I want to see how researchers check assumptions and how they interrupt the results

**Comment by angelinetsui** [comment]
@munozalvalu, one more suggestion. Because the Rmd files are used for showing reviewers and other researchers how we conduct the analysis. I suggest the following: 
1) Move your response to my questions in the issues here, so we all keep a record here.
2) For the assumption and justification of your model and how you run the analyses, put them in the corresponding sections to show reviewers what you are testing and doing at that session. Highlight what the reviewers need to know in order to understand your code.

**Comment by munozalvalu** [comment]
Resoponses to the first set of questions:

1. I explored the distributions for the predictors because I wasn't familiar with the data set and I needed to see how it looks in case there is anything unexpected. The only issue that may be specific to the beta regression would be if the percentiles weren't computed properly. There are no special considerations for beta regressions in terms of the distributions of the predictors.

2. Yes the main assumption specific to beta-regression is that the variance and the mean of the residuals are roughly the same. We test this by looking at overdispersion. The model doesn't show evidence of overdispersion, though it does show evidence of underdipersion which may indicate overly conservative p values. Though this doesn't really play a role because the full-null comparison was not significant. Testing the stability of the model is not unique to beta-regression analyses. It is a way to measure the influence of specific levels of random effects on the results of the model. If removing certain levels makes massive changes in the results then we would have an unstable model, this was not the case in this model.

3. The full vs pruned model was something that was decided within the paper. I didn't make this decision. I just ran the closest model possible to the original planned analyses, which had the same fixed and mixed effects as the pruned model. I only report the pruned model, I did not run the full model described above. However, I do refer to a "full" model in my code, but this is just a naming convention I use. Only the pruned model is being reported on.

**Comment by munozalvalu** [comment]
### Methods:
**Percentile Calculations.**

Raw scorings from the CDI were converted into percentile rankings normed to the individual’s country and age in days. The procedure is outlined in Kartushina et al., (2021). The percentile scores for vocabulary scores accounts for the differences in sizes of CDIs across languages and generally different distributions of language development across countries.  

(This is a bit more of a detailed explanation)

We created percentile vocabulary scores. This was done using the country’s own norms of performance for the specific age group the CDIs were designed to be used for. Most norming data only collected the child’s age in months but due to the rapid development of vocabulary sizes at this stage of life comparing one child that is 12 months and 30 days vs 12 months and 1 day can have quite large differences. To get access to a more granular progression from the norms we performed a quantile regression for each country’s norming data. This was done on a 1 percent quantile. Giving us rankings from 1 to 100. Each of these rankings also interpolated data from months to days by dividing each month by the average length of a month in days (30.457 days). We were then able to fit the raw score from our participants to the raw score of a child to the closest age in days. The score within that age in day was then assigned as the participant’s percentile ranking. Participants outside the age range used for the CDI were removed from further analysis if they were too old or too young when measured with the tool. We also added 15 days to the reported age of the child to account for differences in the reporting of age in months for the infants. This was not added for the Norwegian infants because the norms were already derived with daily age being reported.
Analyses

**Sample size and Removed incomplete data**

The final sample consisted of a total sample of 630 observations with 452 infants from 19 labs using 3 different methods. 68 observations were removed from the initial sample due to incomplete data. 

**Model description**

vocabulary_percentile ~ z.ids_pref * z.test_age + z.ids_pref * cdi.age_range + z.ids_pref * protocol + z.ids_pref * nae + gender + (1 |lab) + (1 | participant)

A linear regression was used to fit the model with a beta error distribution and logit link function for the main exploratory analysis (McCullagh and Nelder 1989; Bolker 2008). The model structure for fixed effects and mixed effects is identical to the model described in the original pre-registered analyses.

**Full-null model comparison**

To test for the overall effect of IDS preference we performed a comparison between a “full” model with all the variables of interest and a “null” model, which excluded IDS preference and all its interactions. This helps to reduce the potential for an artificially inflated Type 1 error rate (Schielzeth, 2011). All significance reports are based on likelihood ratio tests (Dobson, 2002). 

**Data transformation**

IDS preference and test_age were z-transformed. This centered the variables at their mean value and changed it so a 1 unit increase in the variable would be equivalent to an increase of one standard deviation in the variable. This was done to allow for easier interpretation of the model estimates (Schielzeth, 2010) and to make it more likely for the model to converge.

**Model stability**

The stability of the model was evaluated by removing one level of the random effects and comparing the model estimates to the full model. This was repeated for each level of the random effects. The results suggest that the model was stable (see results). 

**Model assumptions**

The assumption of the model lacking overdispersion was tested. Results indicated that it is not overdispersed (dispersion parameter: 0.5177).

**Testing for Collinearity**

Collinearity of the predictors in the model was tested with a Variance Inflation Factor (VIF; Field 2005) which was derived from a linear model without the interaction and random effects. There was no evidence of collinearity between any of the predictors (maximum VIF was 1.17).

**What software was used.**

The model was fitted with R (4.0.2; R Core Team 2020), with the function glmmTMB in the package of the same name (1.1.2; Brooks et al., 2017). To evaluate model assumptions the functions “vif” (R package ‘car’; version 3.0-12; Fox and Weisberg, 2011). The full-null model comparison was performed with the function “anova”(R package “lmtest”; version 0.9.40; Zeileis and Hothorn 2002).

### Results:

**Results for the full-null model comparison**

There was no evidence of an effect of IDS preference (full-null model comparison, likelihood ratio test: χ2 = 4.352, df = 6, p = 0.629). None of the interaction terms involving IDS preference showed evidence of significance either. Gender and North American English were significant, but due to their inclusion in a non-significant interaction the main effects have a limited interpretation and were not further explored. 

### References

1.	Quinn, G. P. and Keough, M. J. (2002). Experimental Designs and Data Analysis for Biologists. Cambridge University Press, Cambridge.
2.	Brooks, M. E., Kristensen, K., van Benthem, K. J., Magnusson, A., Berg, C. W., Nielsen, A., Skaug, H. J., Maechler, M., and Bolker, B. M. (2017). glmmTMB balances speed and flexibility among packages for zero-inflated generalized linear mixed modeling. The R Journal, 9:378–400.
3.	Fox, J. and Weisberg, S. (2011). An R Companion to Applied Regression. Sage, Thousand Oaks CA, second edition.
4.	Dobson, A. J. (2002). An Introduction to Generalized Linear Models. Chapman Hall/CRC, Boca Raton.
5.	Schielzeth, H. and Forstmeier, W. (2009). Conclusions beyond support: overconfident estimates in mixed models. Behavioral Ecololgy, 20:416–420.
6.	Schielzeth, H. (2010). Simple means to improve the interpretability of regression coefficients. Methods in Ecology and Evolution, 1:103–113.
7.	McCullagh, P. and Nelder, J. A. (1989). Generalized linear models. Chapman and Hall, London.
8.	Bolker, B. M., Brooks, M. E., Clark, C. J., Geange, S. W., Poulsen, J. R., Stevens, M. H. H., and White, J.-S. S. (2009). Generalized linear mixed models: a practical guide for ecology and evolution. Trends in Ecolology and Evolution, 24:127–135.
9.	Forstmeier, W. and Schielzeth, H. (2011). Cryptic multiple hypotheses testing in linear models: overestimated effect sizes and the winner’s curse. Behavioural Ecology and Sociobiology, 65:47–55.
10.	Smithson, M., & Verkuilen, J. (2006). A better lemon squeezer? Maximum-likelihood regression with beta-distributed dependent variables. Psychological Methods, 11(1), 54–71. [https://doi.org/10.1037/1082-989X.11.1.54](https://psycnet.apa.org/doi/10.1037/1082-989X.11.1.54)
11.	Achim Zeileis, Torsten Hothorn (2002). Diagnostic Checking in Regression Relationships. R News 2(3), 7-10. URL: https://cran.r-project.org/doc/Rnews/
12.	Kartushina, N., Mani, N., AKTAN-ERCIYES, A. S. L. I., Alaslani, K., Aldrich, N. J., Almohammadi, A., Alroqi, H., Anderson, L., Andonova, E., Aussems, S., Babineau, M., Barokova, M., Bergmann, C., Cashon, C. H., Custode, S., de Carvalho, A., Dimitrova, N., Dynak, A., Farah, R., … Mayor, J. (2021). Covid-19 first lockdown as a window into language acquisition: Associations between caregiver-child activities and vocabulary gains. https://doi.org/10.31234/osf.io/5ejwu

**Comment by munozalvalu** [comment]
Above I've attached a rough draft I sent to Melanie of the write up for what I did. It contains references and brief explanations. As far as I'm aware, beta-regression doesn't have fundamentally different considerations from other generalized linear regression models. The main consideration is of overdispersion and the requirement that the dependent variable is between 0 and 1. Their unique component is that we are assuming the residuals have a beta distribution rather than a gaussian one. This is essentially mandatory when working with percentiles or ratio data because these types of data don't often meet the assumption of homoscedasticity of the residuals or that they are normally distributed. Additionally,  we limit the space of the model fit with the logit function. This makes it so the fitted model can only produce values between 0 and 1 similar to our dependent variable.
